dhcpv4/option_domain_name_server.go

```diff
@@ -0,0 +1,71 @@
+package dhcpv4
+
+import (
+	"fmt"
+	"net"
+)
+
+// This option implements the domain name server option
+// https://tools.ietf.org/html/rfc2132
+
+// OptDomainNameServer represents an option encapsulating the server identifier.
+type OptDomainNameServer struct {
+	NameServers []net.IP
+}
+
+// ParseOptDomainNameServer returns a new OptDomainNameServer from a byte
+// stream, or error if any.
+func ParseOptDomainNameServer(data []byte) (*OptDomainNameServer, error) {
+	if len(data) < 2 {
+		return nil, ErrShortByteStream
+	}
+	code := OptionCode(data[0])
+	if code != OptionDomainNameServer {
+		return nil, fmt.Errorf("expected code %v, got %v", OptionDomainNameServer, code)
+	}
+	length := int(data[1])
+	if length%4 != 0 {
```

Comment:
I think we should also check if the lenght is > 0, the minimun length should be 4.

Reply:
you are right - according to RFC2132 par3.8 the minimum length has to be four octets (one NS)